src/integration-test/java/com/commercetools/sync/integration/sdk2/ctpprojectsource/states/StateSyncIT.java

```diff
@@ -0,0 +1,666 @@
+import static com.commercetools.sync.integration.sdk2.commons.utils.ITUtils.*;
+import static com.commercetools.sync.integration.sdk2.commons.utils.StateITUtils.deleteStatesFromTarget;
+import static com.commercetools.sync.integration.sdk2.commons.utils.StateITUtils.getStateByKey;
+import static com.commercetools.sync.integration.sdk2.commons.utils.TestClientUtils.CTP_SOURCE_CLIENT;
+import static com.commercetools.sync.integration.sdk2.commons.utils.TestClientUtils.CTP_TARGET_CLIENT;
+import static com.commercetools.sync.sdk2.services.impl.UnresolvedReferencesServiceImpl.CUSTOM_OBJECT_TRANSITION_CONTAINER_KEY;
+import static java.lang.String.format;
+import static java.util.concurrent.ThreadLocalRandom.current;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.commercetools.api.client.ProjectApiRoot;
+import com.commercetools.api.defaultconfig.ApiRootBuilder;
+import com.commercetools.api.models.state.State;
+import com.commercetools.api.models.state.StateDraft;
+import com.commercetools.api.models.state.StateDraftBuilder;
+import com.commercetools.api.models.state.StateResourceIdentifier;
+import com.commercetools.api.models.state.StateResourceIdentifierBuilder;
+import com.commercetools.api.models.state.StateRoleEnum;
+import com.commercetools.api.models.state.StateTypeEnum;
+import com.commercetools.sync.sdk2.commons.asserts.statistics.AssertionsForStatistics;
+import com.commercetools.sync.sdk2.commons.models.WaitingToBeResolvedTransitions;
+import com.commercetools.sync.sdk2.commons.utils.CaffeineReferenceIdToKeyCacheImpl;
+import com.commercetools.sync.sdk2.commons.utils.ReferenceIdToKeyCache;
+import com.commercetools.sync.sdk2.services.UnresolvedReferencesService;
+import com.commercetools.sync.sdk2.services.impl.UnresolvedReferencesServiceImpl;
+import com.commercetools.sync.sdk2.states.StateSync;
+import com.commercetools.sync.sdk2.states.StateSyncOptions;
+import com.commercetools.sync.sdk2.states.StateSyncOptionsBuilder;
+import com.commercetools.sync.sdk2.states.helpers.StateSyncStatistics;
+import com.commercetools.sync.sdk2.states.utils.StateTransformUtils;
+import io.vrap.rmf.base.client.ApiHttpMethod;
+import io.vrap.rmf.base.client.ApiHttpResponse;
+import io.vrap.rmf.base.client.utils.CompletableFutureUtils;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class StateSyncIT {
+
+  private String keyA;
+  private String keyB;
+  private String keyC;
+
+  private List<String> errorCallBackMessages;
+  private List<String> warningCallBackMessages;
+  private List<Throwable> errorCallBackExceptions;
+  private ReferenceIdToKeyCache referenceIdToKeyCache;
+
+  @AfterAll
+  static void tearDown() {
+    deleteStatesFromTarget();
```

Comment:
here you need to cleanup source too :)

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/1043/commits/85e629e0636557ef91a6979bd713881a7d798e77